### PR TITLE
fix: Table `azure_network_virtual_networks` column types changed

### LINKED ISSUE #77: azure_network_virtual_networks ip allocations relation can be string array
VirtualNetworkIpAllocations can be string array instead of relation with single field

### PR DESCRIPTION
**Pull request by amanenk**
Column `dhcp_options_dns_servers` of `azure_network_virtual_networks` is now inet array.
Table `azure_network_virtual_network_ip_allocations` is now column of azure_network_virtual_networks.
closes #77

### CODE REVIEW

**Review by roneli — APPROVED** [review]
LGTM, did we test the migration works? we are missing migration tests so verify it manually.

File: resources/network_virtual_networks.go [hunk 1]
```diff
@@ -25,107 +26,113 @@ func NetworkVirtualNetworks() *schema.Table {
...
 				Name:        "bgp_communities_regional_community",
-				Description: "The BGP community associated with the region of the virtual network",
+				Description: "The BGP community associated with the region of the virtual network.",
 				Type:        schema.TypeString,
 				Resolver:    schema.PathResolver("VirtualNetworkPropertiesFormat.BgpCommunities.RegionalCommunity"),
 			},
+			{
+				Name:        "ip_allocations",
+				Description: "Array of IpAllocation which reference this VNET.",
+				Type:        schema.TypeStringArray,
```
Comment (roneli) [inline resources/network_virtual_networks.go:96]: why not type IP Array?

**Review by roneli — CHANGES_REQUESTED** [review]
We should prefer to use INET/MAC/CIDR types for address etc' than text

File: resources/network_virtual_networks.go (hunk 1 above)
Comment (fdistorted) [reply resources/network_virtual_networks.go:96]: this is a list of IDs

**Comment by fdistorted** [comment]
I have tested migrations manually. They work. Agree, we need to add migrations test

File: resources/network_virtual_networks.go [hunk 2]
```diff
@@ -427,27 +416,58 @@ func fetchNetworkVirtualNetworks(ctx context.Context, meta schema.ClientMeta, pa
 	}
 	return nil
 }
-func fetchNetworkVirtualNetworkSubnets(ctx context.Context, meta schema.ClientMeta, parent *schema.Resource, res chan interface{}) error {
-	vn := parent.Item.(network.VirtualNetwork)
+func resolveNetworkVirtualNetworksDhcpOptionsDnsServers(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
```
Comment (roneli) [inline resources/network_virtual_networks.go:419]: SDK v0.5.2 has a InetArray resolver I suggest maybe to use that?

**Review by roneli — CHANGES_REQUESTED** [review]
Added a change suggestion

File: resources/network_virtual_networks.go (hunk 2 above)
Comment (amanenk) [reply resources/network_virtual_networks.go:419]: It seems that resolver does not handle strings. it returns `dhcp_options_dns_servers expected TypeInetArray got *string `

**Review by roneli — APPROVED** [review]
LGTM, requires to fix conflicts first

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**